### ISSUE TITLE
Pressing backspace in voice 2 corrupts the measure

### ISSUE THREAD

**Issue by marczellm** [musescore/MuseScore#17926]
### Issue type

File corruption

### Bug description

Deleting 2 notes from voice 2 by pressing backspace twice results in a state where voice 2 is one 16th shorter than it should across the measure.

### Steps to reproduce

Step 1
![image](https://github.com/musescore/MuseScore/assets/6771275/4ddcc469-4c3b-4723-80f2-bef5c7f1b980)
Step 2: Press backspace twice. The first keypress correctly replaces the last entered note with a rest, but the second wrongly deletes said rest. Now voice 2 is only 15 16ths long.
![image](https://github.com/musescore/MuseScore/assets/6771275/f53f2021-fe6b-4aa9-bffa-b2db8670c9a0)

### MuseScore Version

OS: Windows 10 Version 2009, Arch.: x86_64, MuseScore version (64-bit): 4.0.2-230651553, revision: github-musescore-musescore-dbe7c6d

### Regression

Yes, this used to work in MuseScore 3.x and now is broken

### Operating system

Windows 10

### Additional context

Might be related to https://github.com/musescore/MuseScore/issues/16841 except no saving and loading is involved.

**Comment by MarcSabatella** [comment]
It is by design that voice 2 rests can be deleted - this doesn’t create corruption, as voices other than 1 are OK to have incomplete.  It’s possible that design might change someday, but anyhow, for now there is nothing specifically wrong with the fact that you can delete rests.

What is also by design is that backspace currently simply works like delete rather than delete plus cursor backwards as you might expect.  It’s never done what you might expect, though.  In the past backspace actually acted like undo which was not ideal either.

Making backspace work more as expected
Is an open feature request already - see for example https://github.com/musescore/MuseScore/issues/17557

**Comment by MarcSabatella** [comment]
Yes, you can, but it’s tricky.  One way is to exchange voice 1 & 2 which will cause the gap to be filled with a new rest, then swap back.  But this is part of why I do think there is room for improvement in the design and we should in the future consider disallowing deletion of rests.

But again, it’s not corruption.  Voice 2 rests are deletae, by design.  It’s definitely considering changing that design, but if so it should be a new issue, because it really has nothing to do with backspace specifically.  If we ever do decide to disallow deleting voice 2 rests, it would need to apply to the regular delete command as well.

**Comment by cbjeukendrup** [comment]
That's strange, I get a different result. What do you do exactly?

https://github.com/musescore/MuseScore/assets/48658420/8b11bf54-863e-47f2-b133-5a7192755e49

**Comment by cbjeukendrup** [comment]
Anyway, I can confirm that there is a UI problem here, namely that you cannot enter notes in the place of a deleted rest _using the keyboard_. Using the mouse works well. In this video you see that when using the mouse, the ghost note can snap to the position of the non-existent rest, but the cursor can only snap to notes and rests (and the same issue applies if notes in voice 1 are present).

https://github.com/musescore/MuseScore/assets/48658420/28773c59-3b9d-405b-8712-2fd707c1d5ec

**Comment by marczellm** [comment]
@cbjeukendrup Thanks for looking into it. That seems exactly my problem.

**Comment by cbjeukendrup** [comment]
I've created a new issue for it: #17931 (that seemed better to me than reopening and rewriting this one)
